R/create_credentials.R

```diff
@@ -0,0 +1,233 @@
+#' Create a file with email access credentials
+#'
+#' We can create a file with SMTP configuration and access credentials for the
+#' purpose of more easily sending email messages through [smtp_send()]. With
+#' this file produced, the credentials helper [creds_file()] can be used in the
+#' `credentials` argument of [smtp_send()].
+#'
+#' @param id An option to specify an identifying label for the filename. If no
+#'   `id` is provided, an autogenerated filename will be constructed in the
+#'   following way: `blastula-v1-<host_name>`.
+#' @param provider An optional email provider shortname for autocompleting STMP
+#'   configuration details (the `host`, `port`, `use_ssl` options). Options
+#'   currently include `gmail`, `outlook`, and `office365`. If nothing is
+#'   provided then values for `host`, `port`, and `use_ssl` are expected.
+#' @param user The username for the email account. Typically, this is the email
+#'   address associated with the account.
+#' @param host,port,use_ssl Configuration info for the SMTP server. The `host`
+#'   and `port` parameters are the address and port for the SMTP server;
+#'   `use_ssl` is an option as to whether to use SSL: supply a `TRUE` or `FALSE`
+#'   value.
+#' @param sender_name An option to specify a sender name. This isn't always
+#'   visible to the recipient, however, as some SMTP servers will suppress this.
+#' @examples
+#' \dontrun{
+#' # Create a credentials file to make it
+#' # much easier to send email out through
+#' # Gmail with `smtp_send()`; provide the
+#' # `id` of "gmail_creds"
+#' create_smtp_creds_file(
+#'   id = "gmail_creds",
+#'   provider = "gmail"
+#'   user = "user_name@gmail.com",
+#'   )
+#' }
+#' @export
+create_smtp_creds_file <- function(id = NULL,
+                                   provider = NULL,
+                                   user,
```

Comment:
Move `user` to be the second argument, since it's required and `provider` is not.

Reply:
Done.